cpp/unreal_plugins/UrdfBot/Source/UrdfParser.h

```diff
@@ -0,0 +1,174 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#pragma once
+
+#include <map>
+#include <vector>
+
+#include <Eigen/Dense>
+
+#include <CoreMinimal.h>
+
+class FXmlNode;
+
+struct UrdfJointDesc;
+
+enum class UrdfGeometryType
+{
+    Invalid  = 0,
+    Box      = 1,
+    Sphere   = 2,
+    Cylinder = 3,
+    Mesh     = 4,
+};
+
+enum class UrdfJointType
+{
+    Invalid    = 0,
+    Revolute   = 1,
+    Continuous = 2,
+    Prismatic  = 3,
+    Fixed      = 4,
+    Floating   = 5,
+    Planar     = 6
+};
+
+enum class CalibrationType
+{
+    Invalid = 0,
+    Rising  = 1,
+    Falling = 2
+};
+
+struct UrdfGeometryDesc
+{
+    UrdfGeometryType type_ = UrdfGeometryType::Invalid;
+    FVector size_          = FVector::ZeroVector;
+    float radius_          = 0.0f;
+    float length_          = 0.0f;
+    std::string filename_;
+    float scale_           = 1.0f;
+};
+
+struct UrdfMaterialDesc
+{
+    std::string name_;
+
+    FVector4 color_;
+    std::string texture_;
+
+    // derived data
+    bool is_reference_               = true;
+    UrdfMaterialDesc* material_desc_ = nullptr;
+};
+
+struct UrdfInertialDesc
+{
+    FTransform origin_ = FTransform::Identity;
+    float mass_        = 0.0f;
+    FMatrix inertia_   = FMatrix::Identity;
```

Comment:
I just realized that `inertia_` should default to the zero matrix according to the URDF spec, similar to how `mass_` defaults to zero.

Reply:
fixed

Reply:
This doesn't appear to be fixed. Are all your changes checked in?

Reply:
My bad. checked in latest changes.

Reply:
Just realize that `inertia` is actually required by urdf although we dont use it. Updated the default value anyway.